### PR TITLE
Add Digimon

### PR DESCRIPTION
**Pull request by MikeZeDev**
Mangas and chapters , everything is on same page.
We rely on link anchors & div identifiers .

https://github.com/manga-download/hakuneko/issues/7481

### CODE REVIEW

File: web/src/engine/websites/Digimon.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        const mangaList: Manga[] = [];
+        for (let path of paths) {
+            const data = await FetchCSS<HTMLDivElement>(new Request(new URL(path, this.URI)), 'div.digimoncomic div.pg-container > div[data-inview]');
+            mangaList.push(...data.map(manga => new Manga(this, provider, `${path}#${manga.id}`, manga.querySelector<HTMLHeadingElement>('h3.pg-works_tit__tit').textContent.trim())));
+        };
+        return mangaList;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        return Common.FetchChaptersSinglePageJS.call(this, manga, ChapterScript(manga.Identifier.match(this.mangaIdRegexp)[1]), 2000);
```
Comment (ronny1982) [inline web/src/engine/websites/Digimon.ts:53]: Use a decorator instead and get the identifier from `window.location.hash` within the script itself

File: web/src/engine/websites/Digimon.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaId = url.match(this.mangaIdRegexp)[1];
+        const title = (await FetchCSS<HTMLDivElement>(new Request(new URL(url, this.URI)), `div.digimoncomic div.pg-container > div[id="${mangaId}"] h3.pg-works_tit__tit`)).shift().textContent;
+        return new Manga(this, provider, `${new URL(url).pathname}#${mangaId}`, title.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const paths = ['/digimoncomic/', '/digimoncomic/en/'];
+        const mangaList: Manga[] = [];
+        for (let path of paths) {
```
Comment (ronny1982) [inline web/src/engine/websites/Digimon.ts:45]: Use `const` for variables that are not changed

File: web/src/engine/websites/Digimon.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/digimoncomic/(en/)?#[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaId = url.match(this.mangaIdRegexp)[1];
```
Comment (ronny1982) [inline web/src/engine/websites/Digimon.ts:37]: Construct an `URL` (which is required for the request) and get the manga ID from `url.hash`

File: web/src/engine/websites/Digimon.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    public constructor() {
+        super('digimon', 'Digimon', 'https://digimon.net', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/digimoncomic/(en/)?#[^/]+$`).test(url);
```
Comment (ronny1982) [inline web/src/engine/websites/Digimon.ts:33]: The matcher for the hash in the URL seems to make no sense, should slashes really not be allowed in the hash part of the URL?

File: web/src/engine/websites/Digimon.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        }).filter(element => element.id != window.location.pathname));
+    });
+`;
+}
+
+@Common.PagesSinglePageJS(`[...document.querySelectorAll('div.swiper ul li[aria-label] > img')].reverse().map(image => new URL(image.src, window.location).href);`, 500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    private readonly mangaIdRegexp = new RegExp(/#([^/]+$)/);
```
Comment (ronny1982) [inline web/src/engine/websites/Digimon.ts:22]: The hash from an URL can be extracted from an `URL` object, no need for regex

File: web/src/engine/websites/Digimon.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                title: chapter.text.trim()
+            }
+        }).filter(element => element.id != window.location.pathname));
+    });
+`;
+}
+
+@Common.PagesSinglePageJS(`[...document.querySelectorAll('div.swiper ul li[aria-label] > img')].reverse().map(image => new URL(image.src, window.location).href);`, 500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
```
Comment (ronny1982) [inline web/src/engine/websites/Digimon.ts:21]: There are only a handful of mangas/chapters, is it really worth to add support for this website?

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/Digimon.ts (hunk 6 above)
Comment (MikeZeDev) [reply web/src/engine/websites/Digimon.ts:21]: We have websites with less mangas (for particular reasons ), and more chapters will come i guess🤷

**Pull request merged**